### PR TITLE
feat(uri): accept 'scheme:path' as valid url

### PR DESCRIPTION
**Pull request by sigmaSd**
Unfortunately #19797 is not enough to handle all cases.

Some lsps like (deno) actually do use `scheme:path`

This PR is an attempt to accept `scheme:path` in nvim as valid url which would make it spec compliant.

One note: is `URL_BACKSLASH` still needed or is it possible to remove it? (it doesn't exist in the spec)

### CODE REVIEW

File: src/nvim/testdir/test_gf.vim [hunk 1]
```diff
@@ -75,7 +75,7 @@ func Test_gF()
   edit Xfile1
   call setline(1, ['one', 'Xfile:4', 'three'])
   exe "normal 2G\<C-W>F"
-  call assert_equal('Xfile', bufname('%'))
+  call assert_equal('Xfile:4', bufname('%'))
```
Comment (zeertzjq) [inline src/nvim/testdir/test_gf.vim:78]: Why modify this test?

Comment (zeertzjq) [inline src/nvim/testdir/test_gf.vim:78]: Can you make this test pass without being modified?

**Review by zeertzjq — DISMISSED** [review]

**Comment by sigmaSd** [comment]
I'm currently just modifying tests to see how far can it goes and what else will break

**Comment by sigmaSd** [comment]
Ok the good news is only two tests needs to be disabled

https://github.com/neovim/neovim/pull/19967/files#diff-59d0bc03d4c61e04a234f80256da1c69f33547371df84fc251493fdce2c1e498R62
https://github.com/neovim/neovim/pull/19967/files#diff-59d0bc03d4c61e04a234f80256da1c69f33547371df84fc251493fdce2c1e498R65

These now expect Xfile:number beacuse this:isafile is now a valid url and thus it shows up in bufname

**Comment by sigmaSd** [comment]
@zeertzjq deno is probably going to accept this pr <https://github.com/denoland/deno/pull/16023> which will fix deno on nvim by not using `scheme:path`

So now I don't have practical use case for this anymore.

Should I keep this open or just close it?

**Comment by justinmk** [comment]
> deno is probably going to accept this pr [denoland/deno#16023](https://github.com/denoland/deno/pull/16023) which will fix deno on nvim by not using `scheme:path`
> 
> So now I don't have practical use case for this anymore.
> 
> Should I keep this open or just close it?

Using a colon without slash(es) is a pretty obscure and pedantic way to represent a URI. 

However, we could try this on master for a bit and see what happens. If there are no objections please remind me in a week or so.

File: src/nvim/testdir/test_gf.vim [hunk 2]
```diff
@@ -75,8 +77,10 @@ func Test_gF()
   edit Xfile1
   call setline(1, ['one', 'Xfile:4', 'three'])
   exe "normal 2G\<C-W>F"
-  call assert_equal('Xfile', bufname('%'))
-  call assert_equal(4, getcurpos()[1])
+ "FIXME
+ "call assert_equal('Xfile', bufname('%'))
+ "FIXME
+ "call assert_equal(4, getcurpos()[1])
```
Comment (justinmk) [inline src/nvim/testdir/test_gf.vim:83]: This is a somewhat important feature of vim. Perhaps `path_is_url` can check if the part immediately after the colon is a number? If deno is using `deno:<number>`, well...

**Comment by sigmaSd** [comment]
https://github.com/denoland/deno/pull/16023 was merged and I tested and it works correctly with nvim.

I think I'll just close this it seems too much of a trouble now with no clear benefit.

thanks for your feedback

**Pull request closed without merging**